### PR TITLE
Expand CODEOWNERS for sig-content

### PR DESCRIPTION
**Pull request by NicholasVanSickle**
- Blanket move AzToolsFramework to sig-content, as it's a strictly tool-only module and routing thusly both simplifies existing rules and made routing for the Editor & Prefab subsections easier. If sig-core strongly objects they can also be tagged on this, but I believe it's inappropriately wide to send all AZTF changes to sig-core.
- Also do the same for /Code/Tools/ - let me know if there are non-content tools and I can restore this
- Add Game Objects area covering the prefab & spawnable systems
- Add Editor area covering Editor & Editor scripting integration
- Add UI Editor section for LyShine
- Add scripting area to cover ScriptCanvas

Signed-off-by: Nicholas Van Sickle <nvsickle@amazon.com>

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -37,32 +36,48 @@
 /SerializeContextAnalysis.bat @o3de/sig-core-reviewers @o3de/sig-core-maintainers
 /Tools/SerializeContextAnalyzer/ @o3de/sig-core-reviewers @o3de/sig-core-maintainers
 
+# SIG-Content general ownership area
+/Code/Framework/AzToolsFramework/ @o3de/sig-content-maintainers
+/Code/Tools/ @o3de/sig-content-maintainers
```
Comment (lumberyard-employee-dm) [inline .github/CODEOWNERS:41]: This overlaps the AzTestRunner, but it is fine with me.

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -37,32 +36,48 @@
...
+
+# SIG-Content Editor ownership area
+/Code/Framework/AzFramework/AzFramework/DocumentPropertyEditor/ @o3de/sig-content-maintainers
+/Code/Framework/AzQtComponents/ @o3de/sig-content-maintainers
+/Code/Editor/ @o3de/sig-content-maintainers
+/Gems/EditorPythonBindings/ @o3de/sig-content-maintainers
+/Gems/LandscapeCanvas/ @o3de/sig-content-maintainers
+/Gems/QtForPython/ @o3de/sig-content-maintainers
+/AutomatedTesting/Gem/PythonTests/editor*/ @o3de/sig-content-maintainers
+/AutomatedTesting/Gem/PythonTests/Editor*/ @o3de/sig-content-maintainers
```
Comment (lumberyard-employee-dm) [inline .github/CODEOWNERS:72]: How about GraphCanvas and the GraphModel gems as well.

**Review by gadams3 — APPROVED** [review]

**Review by AMZN-mnaumov — APPROVED** [review]

**Review by lumberyard-employee-dm — APPROVED** [review]

**Review by hultonha — APPROVED** [review]
Updates look good to me, just a quick question, is there a reason why we don't have `@o3de/sig-content-reviewers` and only `@o3de/sig-content-maintainers`? Otherwise all good 👍

**Comment by NicholasVanSickle** [comment]
I checked with the sig-chairs (@monroegm and @lsemp3d) who weren't aware of exactly why the reviewers list doesn't exist for content - I'll try to chase it up later and see if we need another list, but I'm going to merge this for now to at least get better routing.

**Pull request merged**